python/tank/util/filesystem.py

```diff
@@ -343,3 +343,58 @@ def create_valid_filename(value):
         # so that we return a string
         u_src = value.decode("utf-8")
         return exp.sub("_", u_src).encode("utf-8")
+
+def get_permissions(path):
+    """
+    Retrieve the file system permissions for the file or folder in the
+    given path.
+
+    :param filename: Path to the file to be queried for permissions
+    :returns: permissions bits of the file 
+    :raises: OSError - if there was a problem retrieving permissions for the path
+    """
+    return stat.S_IMODE(os.lstat(path)[stat.ST_MODE])
+
+_delete_folder_success = True # global for function below
```

Comment:
we should avoid global variables - what if the method was executed by two worker threads?